### PR TITLE
convert chatGPT to phoenix

### PR DESCRIPTION
**Pull request by Zynton**
Note: watch the todo for the powerbox: `isDisabled: () => !this.odooEditor.isSelectionInBlockRoot()`.

### CODE REVIEW

File: addons/html_editor/static/tests/chatgpt.test.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+const PROMPT_DIALOG_TITLE = "Generate Text with AI";
+const ALTERNATIVES_DIALOG_TITLE = "AI Copywriter";
+
+test("ChatGPT dialog opens in prompt mode when selection is collapsed (with Powerbox)", async () => {
+    const { editor } = await setupEditor("<p>te[]st</p>", { config: { Plugins: EXTRA_PLUGINS } });
+
+    // Select ChatGPT in the Powerbox.
+    insertText(editor, "/");
+    insertText(editor, "ChatGPT");
```
Comment (FrancoisGe) [inline addons/html_editor/static/tests/chatgpt.test.js]:
```suggestion
    insertText(editor, "/ChatGPT");
```

File: addons/html_editor/static/tests/chatgpt.test.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    await animationFrame();
+    press("Enter");
+
+    // Expect the ChatGPT Prompt Dialog to be open.
+    const promptDialogHeaderSelector = `.o_dialog .modal-header:contains("${PROMPT_DIALOG_TITLE}")`
+    await waitFor(promptDialogHeaderSelector);
+    expect(promptDialogHeaderSelector).toHaveCount(1);
```
Comment (FrancoisGe) [inline addons/html_editor/static/tests/chatgpt.test.js]: it test the same thing. waitFor will crash if promptDialogHeaderSelector don t have a match

File: addons/html_editor/static/tests/chatgpt.test.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+test("ChatGPT dialog opens in alternatives mode when selection is not collapsed (with toolbar)", async () => {
+    const { editor } = await setupEditor("<p>te[s]t</p>", { config: { Plugins: EXTRA_PLUGINS } });
+
+    // Select ChatGPT in the toolbar.
+    await contains(".o-we-toolbar [name='ai'] .btn").click();
+
+    // Expect the ChatGPT Alternatives Dialog to be open.
+    const alternativesDialogHeaderSelector = `.o_dialog .modal-header:contains("${ALTERNATIVES_DIALOG_TITLE}")`
+    await waitFor(alternativesDialogHeaderSelector);
+    expect(alternativesDialogHeaderSelector).toHaveCount(1);
```
Comment (FrancoisGe) [inline addons/html_editor/static/tests/chatgpt.test.js]: same

File: addons/html_editor/static/tests/chatgpt.test.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    await waitFor(promptDialogHeaderSelector);
+    expect(promptDialogHeaderSelector).toHaveCount(1);
+
+    // Expect the ChatGPT Alternatives Dialog not to be open.
+    const alternativesDialogHeaderSelector = `.o_dialog .modal-header:contains("${ALTERNATIVES_DIALOG_TITLE}")`
+    expect(alternativesDialogHeaderSelector).toHaveCount(0);
+});
+
+test("ChatGPT dialog opens in alternatives mode when selection is not collapsed (with toolbar)", async () => {
+    const { editor } = await setupEditor("<p>te[s]t</p>", { config: { Plugins: EXTRA_PLUGINS } });
```
Comment (FrancoisGe) [inline addons/html_editor/static/tests/chatgpt.test.js]: Maybe just use MAIN_PLUGIN + chatGPT_Plugin ? You don t need to have all the other extra plugin

File: addons/html_editor/static/tests/chatgpt.test.js (hunk 1 above)
Comment (Zynton) [reply addons/html_editor/static/tests/chatgpt.test.js]: I saw this done in another test, I assumed there was a reason for it. Good to know there wasn't!

File: addons/html_editor/static/tests/chatgpt.test.js (hunk 2 above)
Comment (Zynton) [reply addons/html_editor/static/tests/chatgpt.test.js]: Do you mean I should remove the `expect` then?

File: addons/html_editor/static/src/others/chatgpt_plugin.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        powerboxCategory: { id: "ai", name: _t("AI Tools"), sequence: 50 },
+        powerboxCommands: {
+            name: _t("ChatGPT"),
+            description: _t("Generate or transform content with AI."),
+            category: "ai",
+            fontawesome: "fa-magic",
+            action(dispatch) {
+                dispatch("OPEN_CHATGPT_DIALOG");
+            },
+            // isDisabled: () => !this.odooEditor.isSelectionInBlockRoot(), // TODO!
```
Comment (Zynton) [inline addons/html_editor/static/src/others/chatgpt_plugin.js:33]: @ged-odoo I couldn't find anything in Phoenix for this. Does it exist or not yet?

File: addons/html_editor/static/tests/chatgpt.test.js (hunk 2 above)
Comment (FrancoisGe) [reply addons/html_editor/static/tests/chatgpt.test.js]: yes

File: addons/html_editor/static/src/others/chatgpt_alternatives_dialog.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
+import { useState, status } from "@odoo/owl";
+import { ChatGPTDialog } from "./chatgpt_dialog";
+
+export const DEFAULT_ALTERNATIVES_MODES = {
+    correct: 'Correct',
```
Comment (ged-odoo) [inline addons/html_editor/static/src/others/chatgpt_alternatives_dialog.js]: shouldn't this be translated? (import `_t`)

Comment (Zynton) [reply addons/html_editor/static/src/others/chatgpt_alternatives_dialog.js]: I believe you're absolutely correct ^^

File: addons/html_editor/static/src/others/chatgpt_plugin.js (hunk 5 above)
Comment (ged-odoo) [reply addons/html_editor/static/src/others/chatgpt_plugin.js:33]: not yet! i wasn't aware there was any powerbox commands that could be disabled

**Pull request merged**